handlers/handlers.go

```diff
@@ -66,15 +116,49 @@ func (i *Impl) GET_SensorData_ID(rw http.ResponseWriter, req *http.Request, ps h
 	rw.Write(js)
 }
 
-// [GET] /api/sensor_data
-func (i *Impl) SensorData(rw http.ResponseWriter, req *http.Request, _ httprouter.Params) {
-	if req.Method == "GET" {
-		js, err := json.Marshal(i.sd_table)
-		if err != nil {
-			http.Error(rw, err.Error(), http.StatusInternalServerError)
-			panic(err)
-		}
-		rw.Header().Set("Content-Type", "application/json")
-		rw.Write(js)
+func (i *Impl) GET_SensorData(rw http.ResponseWriter, req *http.Request, _ httprouter.Params) {
+	i.mu.Lock()
+	js, err := json.Marshal(i.sd_table)
+	i.mu.Unlock()
+	if err != nil {
+		http.Error(rw, err.Error(), http.StatusInternalServerError)
+		panic(err)
+	}
+	rw.Header().Set("Content-Type", "application/json")
+	rw.Write(js)
+}
+
+func (i *Impl) GET_Faults(rw http.ResponseWriter, req *http.Request, _ httprouter.Params) {
+	i.mu.Lock()
+	js, err := json.Marshal(i.faults)
+	i.mu.Unlock()
+	if err != nil {
+		http.Error(rw, err.Error(), http.StatusInternalServerError)
+		panic(err)
 	}
+
+	rw.Header().Set("Content-Type", "application/json")
+	rw.Write(js)
+}
+
+func (i *Impl) DELETE_Faults_ID(rw http.ResponseWriter, req *http.Request, ps httprouter.Params) {
+	sensor_id := ps.ByName("sensor_id")
+
+	i.mu.Lock()
+	flts, exist := i.faults[sensor_id]
+	if !exist {
+		rw.WriteHeader(404)
+		rw.Write([]byte("Can't find fault data for the id specified"))
+		i.mu.Unlock()
+		return
+	}
+	js, err := json.Marshal(flts)
```

Comment:
Should we nil the location instead? `i.faults[sensor_id] = nil` ? since this is delete the fault from the table